atoma-state/src/state_manager.rs

```diff
@@ -1473,31 +1451,22 @@ impl AtomaState {
         stack_small_id: i64,
         new_hash: [u8; 32],
     ) -> Result<()> {
-        let mut transaction = self.db.begin().await?;
-
-        let existing_hash = sqlx::query_scalar::<_, Vec<u8>>(
-            "SELECT total_hash, num_total_messages FROM stacks WHERE stack_small_id = ?",
-        )
-        .bind(stack_small_id)
-        .fetch_one(&mut *transaction)
-        .await
-        .map_err(|_| AtomaStateManagerError::StackNotFound)?;
-
-        let total_hash = [existing_hash, new_hash.to_vec()].concat();
-
-        // NOTE: also update the num_total_messages
-        sqlx::query(
+        let rows_affected = sqlx::query(
             "UPDATE stacks 
-                    SET total_hash = ?,
-                    num_total_messages = num_total_messages + 1
-                    WHERE stack_small_id = ?",
+            SET total_hash = total_hash || ?,
```

Comment:
you changed `?` into `$1` etc, in other queries, you want to stay consistent and change this as well